### PR TITLE
访问策略转换为插件时，bk-cors 插件 allow_origins、allow_origins_by_regex 二选一

### PR DESCRIPTION
**Pull request by alex-smile**
### Description

Fixes # (issue)
- 访问策略转换为插件时，bk-cors 插件 allow_origins、allow_origins_by_regex 二选一

### Checklist

- [ ] 填写 PR 描述及相关 issue (write PR description and related issue)
- [ ] 代码风格检查通过 (code style check passed)
- [ ] PR 中包含单元测试 (include unit test)
- [ ] 单元测试通过 (unit test passed)
- [ ] 本地开发联调环境验证通过 (local development environment verification passed)

### CODE REVIEW

**Review by wklken — APPROVED** [review]
LGTM

**Review by zhu327 — APPROVED** [review]

File: src/dashboard/apigateway/apigateway/controller/crds/release_data/access_strategy.py [hunk 1]
```diff
@@ -137,43 +137,49 @@ def _to_plugin_config(self, access_strategy: AccessStrategy) -> Dict[str, Any]:
...
         if allow_origins_by_regex:
             plugin_config["allow_origins_by_regex"] = allow_origins_by_regex
 
         return plugin_config
 
-    def _convert_allowed_origins(self, allowed_origins: List[str]) -> Tuple[str, Optional[List]]:
-        allow_origins = []
+    def _convert_allowed_origins(self, allowed_origins: List[str]) -> Tuple[Optional[str], Optional[List]]:
+        # 存在 "*"，即支持所有域名
+        allow_all_origins = True if "*" in allowed_origins else False
```
Comment (piglei) [inline src/dashboard/apigateway/apigateway/controller/crds/release_data/access_strategy.py:157]: NIt: `allow_all_origins = "*" in allowed_origins`

File: src/dashboard/apigateway/apigateway/controller/crds/release_data/access_strategy.py [hunk 2]
```diff
@@ -137,43 +137,49 @@ def _to_plugin_config(self, access_strategy: AccessStrategy) -> Dict[str, Any]:
...
+        if allow_all_origins:
+            return "**", None
+
+        # 域名中包含类似 http://*.example.com 的情况，则所有域名均应转换为正则模式，
+        # apisix 新版本不允许同时使用 allow_origin, allow_origins_by_regex
+        should_use_regex = bool([origin for origin in allowed_origins if "*" in origin])
```
Comment (piglei) [inline src/dashboard/apigateway/apigateway/controller/crds/release_data/access_strategy.py:163]: Nit: `should_use_regex = any("*" in origin for origin in allowed_origins)`

File: src/dashboard/apigateway/apigateway/controller/crds/release_data/access_strategy.py [hunk 3]
```diff
@@ -137,43 +137,49 @@ def _to_plugin_config(self, access_strategy: AccessStrategy) -> Dict[str, Any]:
...
-            else:
-                allow_origins.append(origin)
-
-        # allow_origins 不能为空字符串，未设置时将使用默认值 "*"，因此为空时将其设置为 null
-        return ",".join(allow_origins) or "null", allow_origins_by_regex
+            # 1. 域名中，可能已包含正则中的特殊字符，需转义，如 "."
+            # 2. 域名中 * 表示任意字符，正则表达式中，需替换为".*"，如 http://*.example.com，需匹配 http://a.example.com, http://b.example.com
+            # 3. 正则添加开头和结尾 "^$"，防止 http://a.example.com1 匹配到正则 "http://.*\.example\.com"
+            origin_by_regex = "^{origin}$".format(
+                origin=origin.replace(".", r"\.")
```
Comment (piglei) [inline src/dashboard/apigateway/apigateway/controller/crds/release_data/access_strategy.py:174]: 可以用 [re.escape](https://docs.python.org/3/library/re.html#re.escape)

**Pull request merged**